endaq/calc/shock.py

```diff
@@ -274,7 +274,88 @@ def amp_factor(damp):
     max_pvss = pvss.max()
     max_f_pvss = pvss.mul(pvss.index, axis=0).max()
 
-    return namedtuple("HalfSinePulseParameters", "amplitude, period")(
+    return HalfSineWavePulse(
         amplitude=2 * np.pi * max_f_pvss,
-        period=max_pvss / (4 * amp_factor(damp) * max_f_pvss),
+        duration=max_pvss / (4 * amp_factor(damp) * max_f_pvss),
     )
+
+
+class HalfSineWavePulse(NamedTuple):
```

Comment:
For clarity, it might be better to put the class definition before the functions that use it.

Reply:
sure, makes no difference to me. done in fbb324d761043597765222510a5de9a5d272025e